src/machine/machine_k210.go

```diff
@@ -0,0 +1,94 @@
+// +build k210
+
+package machine
+
+import (
+	"device/kendryte"
+	"runtime/interrupt"
+)
+
+func CPUFrequency() uint32 {
+	return 390000000
+}
+
+type PinMode uint8
+
+const (
+	PinInput PinMode = iota
+	PinOutput
+	PinPWM
+	PinSPI
+	PinI2C = PinSPI
+)
+
+// Configure this pin with the given configuration.
+func (p Pin) Configure(config PinConfig) {
+}
+
+// Set the pin to high or low.
+func (p Pin) Set(high bool) {
+}
+
+// Get returns the current value of a GPIO pin.
+func (p Pin) Get() bool {
+	return true
+}
+
+type FPIOA struct {
+	Bus *kendryte.FPIOA_Type
+}
+
+var (
+	FPIOA0 = FPIOA{Bus: kendryte.FPIOA}
+)
+
+func (fpioa FPIOA) Init() {
+	// Enable APB0 clock.
+	kendryte.SYSCTL.CLK_EN_CENT.Set(kendryte.SYSCTL_CLK_EN_CENT_APB0_CLK_EN)
+
+	// Enable FPIOA peripheral.
+	kendryte.SYSCTL.CLK_EN_PERI.Set(kendryte.SYSCTL_CLK_EN_PERI_FPIOA_CLK_EN)
```

Comment:
This looks like a job for the runtime to do at reset.

Reply:
Yes this function is called by the runtime in `initPeripherals()`